File: github-api-c#/GitHubTests/GitHub/API/IssuesAPI.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using GitHubTests.Helpers;
using Newtonsoft.Json;

namespace GitHubTests.GitHub.API
{
    class IssuesAPI
    {
        private static string GitHubAPIBaseURL = "https://api.github.com";

        public List<IssueDetails> GetAllIssues(string organization, string repository, string options = null)
        {
            var uri = string.Format("{0}/repos/{1}/{2}/issues", GitHubAPIBaseURL, organization, repository);
            if (options != null)
            {
                uri = uri + "?" + options;
            }
            var message = new HttpRequestMessage(HttpMethod.Get, new Uri(uri));
            var response = HttpHelper.SendRequest(message);
            var content = response.Content.ReadAsStringAsync().Result;
            return JsonConvert.DeserializeObject<List<IssueDetails>>(content);
        }

        public IssueDetails GetIssueDetails(string organization, string repository, int issueNumber)
        {
            var uri = string.Format("{0}/repos/{1}/{2}/issues/{3}", GitHubAPIBaseURL, organization, repository, issueNumber.ToString());
            var message = new HttpRequestMessage(HttpMethod.Get, new Uri(uri));
            var response = HttpHelper.SendRequest(message);
            var content = response.Content.ReadAsStringAsync().Result;
            return JsonConvert.DeserializeObject<IssueDetails>(content);
        }

        public IssueDetails CreateIssue(string organization, string repository, CreateIssueDetails issue)
        {
            var uri = string.Format("{0}/repos/{1}/{2}/issues", GitHubAPIBaseURL, organization, repository);
            var message = new HttpRequestMessage(HttpMethod.Post, new Uri(uri));
            var content = JsonConvert.SerializeObject(issue,
                            new JsonSerializerSettings
                            {
                                NullValueHandling = NullValueHandling.Ignore
                            });
            message.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
            message.Content = new StringContent(content, Encoding.UTF8, "application/json");
            var token = Environment.GetEnvironmentVariable("GITHUB_TOKEN");
            message.Headers.Authorization = new AuthenticationHeaderValue("Token", token);
            var response = HttpHelper.SendRequest(message);
            var result = response.Content.ReadAsStringAsync().Result;
            return JsonConvert.DeserializeObject<IssueDetails>(result);
        }
    }
}

File: typescript/github/api/IssuesApi.ts
import request from "supertest";
import { Settings } from "../Settings";

export class IssuesApi {
    user: string;
    repo: string;
    token: string;

    constructor(user: string, repo: string, token: string) {
        this.user = user;
        this.repo = repo;
        this.token = token;
    }

    async getIssueDetails(number: number): Promise<request.Test> {
        return await request(Settings.API_URL)
            .get(`/repos/${this.user}/${this.repo}/issues/${number}`)
            .set("User-Agent", "supertest")
            .set("Authorization", `Bearer ${Settings.API_TOKEN}`);
    }

    async createIssue(payload: object): Promise<request.Test> {
        return await request(Settings.API_URL)
            .post(`/repos/${this.user}/${this.repo}/issues`)
            .set("User-Agent", "supertest")
            .set("Authorization", `Bearer ${Settings.API_TOKEN}`)
            .set("Content-Type", "application/json")
            .send(payload);
    }
}
